### PR TITLE
Converge Import from Git/Devfile/Dockerfile flows

### PR DESCRIPTION
**Pull request by divyanshiGupta**
**Fixes:**  https://issues.redhat.com/browse/ODC-6154
**Fixes:** https://issues.redhat.com/browse/ODC-6155

### CODE REVIEW

File: frontend/packages/dev-console/src/components/import/ImportStrategySection.tsx [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  } = values;
+  const [alertVariant, setAlertVariant] = React.useState<AlertVariant>(AlertVariant.default);
+  const [alertTitle, setAlertTitle] = React.useState<string>('');
+  const [alertDescription, setAlertDescription] = React.useState<React.ReactNode>('');
+  const [importStrategyComponent, setImportStrategyComponent] = React.useState<React.ReactNode>(
+    null,
+  );
+  const [recommendedValues, setRecommendedValues] = React.useState<any>(null);
+
+  React.useEffect(() => {
```
Comment (sahil143) [inline frontend/packages/dev-console/src/components/import/ImportStrategySection.tsx:43]: I think instead of useEffect and useState here, a useMemo can work here to get the Alert props and component to render

File: frontend/packages/dev-console/src/components/import/ImportStrategySection.tsx [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        {recommendedStrategy &&
+          (!showEditImportStrategy ? (
+            <Button
+              variant={ButtonVariant.link}
+              onClick={() => {
+                setRecommendedValues(values);
+                setFieldValue('import.showEditImportStrategy', true);
+              }}
+              icon={<PencilAltIcon />}
+              style={{ display: 'inline-flex', justifyContent: 'flex-end', paddingRight: '0px' }}
```
Comment (sahil143) [inline frontend/packages/dev-console/src/components/import/ImportStrategySection.tsx:132]: can move the styles to separate scss file

File: frontend/packages/dev-console/src/components/import/ImportStrategySection.tsx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+            </Button>
+          ) : (
+            <Button
+              variant={ButtonVariant.link}
+              onClick={() => {
+                setFieldValue('import.showEditImportStrategy', false);
+                setValues(recommendedValues);
+              }}
+              icon={<UndoIcon />}
+              style={{ display: 'inline-flex', justifyContent: 'flex-end', paddingRight: '0px' }}
```
Comment (sahil143) [inline frontend/packages/dev-console/src/components/import/ImportStrategySection.tsx:144]: same as above

File: frontend/packages/dev-console/src/components/import/ImportStrategySelector.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+import { DetectedStrategyFormData } from './import-types';
+
+const ImportStrategySelector: React.FC = () => {
+  const { t } = useTranslation();
+  const {
+    values: {
+      import: { recommendedStrategy },
+    },
+    // setFieldValue,
+    // setFieldTouched,
```
Comment (sahil143) [inline frontend/packages/dev-console/src/components/import/ImportStrategySelector.tsx:17]: remove comments

File: frontend/packages/dev-console/src/components/import/ImportStrategySelector.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    },
+  ];
+
+  // const onSelect = React.useCallback(
+  //   (item) => {
+  //     setFieldValue('import.selectedStrategy', item);
+  //     setFieldTouched('import.selectedStrategy', true);
+  //   },
+  //   [setFieldTouched, setFieldValue],
+  // );
```
Comment (sahil143) [inline frontend/packages/dev-console/src/components/import/ImportStrategySelector.tsx:56]: cleanup commented code

File: frontend/packages/dev-console/src/components/import/ImportStrategySelector.tsx [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+          <Tile
+            title={item.title}
+            icon={item.icon}
+            // onClick={() => onSelect(item)}
+            isSelected={selected.value.type === item.type}
+            key={item.name}
+            style={{ width: '208px' }}
+          >
+            {recommendedStrategy?.type === item.type && (
+              <span className="odc-selector-card__recommended">
```
Comment (sahil143) [inline frontend/packages/dev-console/src/components/import/ImportStrategySelector.tsx:71]: Where is this class coming from?

File: frontend/packages/dev-console/src/components/import/ImportStrategySelector.tsx [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+    <FormGroup fieldId={fieldId} label={t('devconsole~Import Strategy')}>
+      <div style={{ display: 'flex', justifyContent: 'space-between' }}>
+        {itemList.map((item) => (
+          <Tile
+            title={item.title}
+            icon={item.icon}
+            // onClick={() => onSelect(item)}
+            isSelected={selected.value.type === item.type}
+            key={item.name}
+            style={{ width: '208px' }}
```
Comment (sahil143) [inline frontend/packages/dev-console/src/components/import/ImportStrategySelector.tsx:68]: Any reason for using a specific value for width?

File: frontend/packages/dev-console/src/components/import/builder/BuilderImageTagSelector.tsx [hunk 8]
```diff
@@ -62,14 +68,16 @@ const BuilderImageTagSelector: React.FC<BuilderImageTagSelectorProps> = ({
...
+      <div style={!showEditImportStrategy ? { display: 'none' } : {}}>
+        <DropdownField
+          name="image.tag"
+          label={t('devconsole~Builder Image version')}
+          items={tagItems}
+          title={tagItems[selectedImageTag]}
+          fullWidth
+          required
+        />
+      </div>
```
Comment (sahil143) [inline frontend/packages/dev-console/src/components/import/builder/BuilderImageTagSelector.tsx:80]:
```suggestion
      {showEditImportStrategy &&
        <DropdownField
          name="image.tag"
          label={t('devconsole~Builder Image version')}
          items={tagItems}
          title={tagItems[selectedImageTag]}
          fullWidth
          required
        />
}
```

File: frontend/packages/dev-console/src/components/import/builder/BuilderSection.tsx [hunk 9]
```diff
@@ -1,31 +1,30 @@
...
+    },
+  } = useFormikContext<FormikValues>();
   if (!builderImages) {
     return null;
   }
 
   return (
     <>
-      <FormSection title={t('devconsole~Builder')} fullWidth>
+      <FormSection fullWidth style={!showEditImportStrategy ? { display: 'none' } : {}}>
```
Comment (sahil143) [inline frontend/packages/dev-console/src/components/import/builder/BuilderSection.tsx:27]: same as above. we shouldn't render the FormSection itself instead of setting the style `display: none`

File: frontend/packages/dev-console/src/components/import/devfile/DevfileInfo.tsx [hunk 10]
```diff
@@ -1,28 +1,37 @@
...
-        {iconUrl && (
+        {iconUrl ? (
           <img
             className="co-catalog-item-icon__img co-catalog-item-icon__img--large"
             src={iconUrl}
             alt={displayName}
             aria-hidden
           />
+        ) : (
+          defaultIcon
```
Comment (sahil143) [inline frontend/packages/dev-console/src/components/import/devfile/DevfileInfo.tsx:34]: No need to initialize a constant above
```suggestion
          <LayerGroupIcon size="xl" />
```

**Review by sahil143 — CHANGES_REQUESTED** [review]

**Comment by sahil143** [comment]
@divyanshiGupta found two issues
1. clicking on the title of the recommended image seems to open the section that changes the strategy
2. Clicking on other strategies doesn't seem to update the form

Kapture 2021-08-20 at 20 18 01

File: frontend/packages/dev-console/src/components/import/devfile/DevfileStrategySection.tsx [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+      tags: devfileJSON.metadata?.version ? [devfileJSON.metadata.version] : [],
+      iconClass: devfileJSON.metadata?.name ? `icon-${devfileJSON.metadata?.name}` : '',
+    };
+  };
+
+  const handleDevfileChange = React.useCallback(async () => {
+    const gitService = getGitService(url, type, ref, dir, secretResource, devfile.devfilePath);
+    if (!values.devfile?.devfileSourceUrl) {
+      // No need to check the existence of the file, waste of a call to the gitService for this need
+      const devfileContents = gitService && (await gitService.getDevfileContent());
```
Comment (invincibleJai) [inline frontend/packages/dev-console/src/components/import/devfile/DevfileStrategySection.tsx:34]: will it be safe here to put await in try-catch block?

File: frontend/packages/dev-console/src/components/import/git/GitSection.tsx [hunk 12]
```diff
@@ -114,95 +171,71 @@ const GitSection: React.FC<GitSectionProps> = ({
...
+          type: ImportStrategy.DEVFILE,
+          priority: 2,
+          detectedFiles: [],
+        });
+        setFieldValue('import.recommendedStrategy', null);
+        values.formType !== 'edit' && setFieldValue('import.showEditImportStrategy', true);
       }
 
+      if (importStrategies.length > 0) {
+        switch (importStrategies[0].type) {
```
Comment (invincibleJai) [inline frontend/packages/dev-console/src/components/import/git/GitSection.tsx:193]: sorry, i missed why it is [0]?

File: frontend/packages/dev-console/src/components/import/GitImportForm.tsx [hunk 13]
```diff
@@ -27,16 +28,23 @@ const GitImportForm: React.FC<FormikProps<FormikValues> & GitImportFormProps> =
...
-        <DockerSection buildStrategy={values.build.strategy} />
-        <AppSection
-          project={values.project}
-          noProjectsAvailable={projects.loaded && _.isEmpty(projects.data)}
-        />
-        <ResourceSection />
-        <PipelineSection builderImages={builderImages} />
-        <AdvancedSection values={values} />
+        <GitSection builderImages={builderImages} />
+        {values.git.validated === ValidatedOptions.success && (
```
Comment (rottencandy) [inline frontend/packages/dev-console/src/components/import/GitImportForm.tsx:32]: I think we need an option to show the rest of the form even if the url isn't validated, for eg. when an unsupported repo host is used.

File: frontend/packages/dev-console/src/components/import/ImportStrategySection.tsx [hunk 14]
```diff
@@ -0,0 +1,162 @@
...
+import { NormalizedBuilderImages } from '../../utils/imagestream-utils';
+import BuilderSection from './builder/BuilderSection';
+import DevfileStrategySection from './devfile/DevfileStrategySection';
+import DockerSection from './git/DockerSection';
+import ImportStrategySelector from './ImportStrategySelector';
+import FormSection from './section/FormSection';
+
+export interface ImportStrategySectionProps {
+  builderImages: NormalizedBuilderImages;
+  existingPipeline?: PipelineKind;
```
Comment (rottencandy) [inline frontend/packages/dev-console/src/components/import/ImportStrategySection.tsx:17]: This prop isn't used anywhere.

File: frontend/packages/dev-console/src/components/import/devfile/DevfileStrategySection.tsx (hunk 11 above)
Comment (rottencandy) [reply frontend/packages/dev-console/src/components/import/devfile/DevfileStrategySection.tsx:34]: @invincibleJai the getDevfileContent util already catches fetch errors and returns null if any.

**Comment by rottencandy** [comment]
If I enter a url and then change it, it still shows as validated:

https://user-images.githubusercontent.com/20013884/130280245-27d7a400-cc21-463e-9629-bb56387c15cb.mp4

File: frontend/packages/dev-console/src/components/import/ImportStrategySection.tsx [hunk 15]
```diff
@@ -0,0 +1,162 @@
...
+      selectedStrategy,
+      showEditImportStrategy,
+      recommendedStrategy,
+    },
+    devfile,
+    docker,
+  } = values;
+  const [alertVariant, setAlertVariant] = React.useState<AlertVariant>(AlertVariant.default);
+  const [alertTitle, setAlertTitle] = React.useState<string>('');
+  const [alertDescription, setAlertDescription] = React.useState<React.ReactNode>('');
```
[the remark this answers is not in the document]
Comment (divyanshiGupta) [reply frontend/packages/dev-console/src/components/import/ImportStrategySection.tsx:37]: Using React.ReactNode for description because it excepts both string and  ReactElement. Will update it anyways.

File: frontend/packages/dev-console/src/components/import/builder/BuilderSection.tsx (hunk 9 above)
Comment (divyanshiGupta) [reply frontend/packages/dev-console/src/components/import/builder/BuilderSection.tsx:27]: We need to add this style because we want to render the component as it sets the required values.

File: frontend/packages/dev-console/src/components/import/git/GitSection.tsx (hunk 12 above)
Comment (divyanshiGupta) [reply frontend/packages/dev-console/src/components/import/git/GitSection.tsx:198]: Because the first strategy will be of the highest priority

File: frontend/packages/dev-console/src/components/edit-application/__tests__/edit-application-data.ts [hunk 16]
```diff
@@ -630,6 +633,9 @@ export const externalImageValues: DeployImageFormData = {
   build: { env: [], triggers: {}, strategy: '' },
   isSearchingForImage: false,
   healthChecks: healthChecksProbeInitialData,
+  import: {
+    showEditImportStrategy: true,
+  },
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/edit-application/__tests__/edit-application-data.ts:638]: Why is this needed for deploy image form data?

File: frontend/packages/dev-console/src/components/import/ImportForm.tsx [hunk 17]
```diff
@@ -1,8 +1,10 @@
 import * as React from 'react';
+import { ValidatedOptions } from '@patternfly/react-core';
 import { Formik, FormikProps } from 'formik';
 import * as _ from 'lodash';
 import { useTranslation } from 'react-i18next';
 import { connect } from 'react-redux';
+import { ImportStrategy } from '@console/git-service/src';
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/import/ImportForm.tsx:7]:
```suggestion
import { ImportStrategy } from '@console/git-service';
```

File: frontend/packages/dev-console/src/components/import/ImportStrategySection.tsx [hunk 18]
```diff
@@ -0,0 +1,168 @@
...
+        description = '';
+        variant = AlertVariant.info;
+      }
+      setAlertVariant(variant);
+      setAlertTitle(title);
+      setAlertDescription(description);
+      setImportStrategyComponent(component);
+    } else {
+      setAlertVariant(AlertVariant.warning);
+      setAlertTitle(t('devconsole~Unable to detect build type'));
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/import/ImportStrategySection.tsx:100]: Shouldn't it say unable to detect import strategy?

File: frontend/packages/dev-console/src/components/import/ImportStrategySelector.tsx [hunk 19]
```diff
@@ -0,0 +1,85 @@
...
+const ImportStrategySelector: React.FC = () => {
+  const { t } = useTranslation();
+  const {
+    values: {
+      import: { recommendedStrategy },
+    },
+    // setFieldValue,
+    // setFieldTouched,
+  } = useFormikContext<FormikValues>();
+  const [selected] = useField<DetectedStrategyFormData>('import.selectedStrategy');
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/import/ImportStrategySelector.tsx:20]: You can directly take this out from value above. No need to use `useField` if you just need the value.

File: frontend/packages/dev-console/src/components/import/builder/BuilderImageTagSelector.tsx [hunk 20]
```diff
@@ -62,14 +68,16 @@ const BuilderImageTagSelector: React.FC<BuilderImageTagSelectorProps> = ({
...
     <>
-      <DropdownField
-        name="image.tag"
-        label={t('devconsole~Builder Image version')}
-        items={tagItems}
-        title={tagItems[selectedImageTag]}
-        fullWidth
-        required
-      />
+      <div style={!showEditImportStrategy ? { display: 'none' } : {}}>
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/import/builder/BuilderImageTagSelector.tsx:71]: Why do you need this check here? You should not render the whole `BuildImageSection` component when the strategy is not builder image.

File: frontend/packages/dev-console/src/components/import/builder/BuilderSection.tsx (hunk 9 above)
Comment (rohitkrai03) [reply frontend/packages/dev-console/src/components/import/builder/BuilderSection.tsx:27]: The validation should be updated to not require builder image if strategy is not builder image.

File: frontend/packages/git-service/src/services/bitbucket-service.ts [hunk 21]
```diff
@@ -55,6 +55,8 @@ export class BitbucketService extends BaseService {
       host,
       defaultBranch: this.gitsource.ref || 'HEAD',
       contextDir,
+      devfilePath: this.gitsource.devfilePath,
+      dockerfilePath: this.gitsource.dockerfilePath,
```
Comment (rohitkrai03) [inline frontend/packages/git-service/src/services/bitbucket-service.ts:59]: Can it be combined into one generic property `path` or `filePath`?

File: frontend/packages/git-service/src/services/bitbucket-service.ts [hunk 22]
```diff
@@ -125,13 +127,17 @@ export class BitbucketService extends BaseService {
...
+  getDockerfileContent = () =>
+    this.getFileContent(`${this.metadata.contextDir}/${this.metadata.dockerfilePath}`);
 
-  isDevfilePresent = () => this.isFilePresent(`${this.metadata.contextDir}/devfile.yaml`);
+  isDevfilePresent = () =>
+    this.isFilePresent(`${this.metadata.contextDir}${this.metadata.devfilePath}`);
 
-  getDevfileContent = () => this.getFileContent(`${this.metadata.contextDir}/devfile.yaml`);
+  getDevfileContent = () =>
+    this.getFileContent(`${this.metadata.contextDir}${this.metadata.devfilePath}`);
```
Comment (rohitkrai03) [inline frontend/packages/git-service/src/services/bitbucket-service.ts:141]: These functions can also be combined into one if we use generic `filePath` property.

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]

**Comment by rohitkrai03** [comment]
@divyanshiGupta Tests are failing in git service.

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]
- I am seeing this weird issue where the first validation attempt fails and then succeeds next time. This is happening on S2I form as well.

https://user-images.githubusercontent.com/6041994/130355025-53cbf029-ffc9-4576-be4c-ad20dccfb2d1.mov

- Devfile flow from samples catalog or developer catalog doesn't work anymore.


https://user-images.githubusercontent.com/6041994/130355122-f8083e4d-d9f8-4d3a-b277-2d0ef2868ab5.mov

File: frontend/packages/dev-console/src/components/import/git/GitSection.tsx [hunk 23]
```diff
@@ -87,22 +135,39 @@ const GitSection: React.FC<GitSectionProps> = ({
...
+        gitProvider,
+      } = importStrategyData;
+
+      const gitType = gitTypeTouched ? values.git.type : gitProvider;
       const gitRepoName = detectGitRepoName(url);
-      const showGitType = gitType === GitTypes.unsure || (gitTypeTouched && !isSubmitting);
+      const showGitType = gitProvider === GitProvider.UNSURE || (gitTypeTouched && !isSubmitting);
 
       setFieldValue('git.type', gitType);
       setFieldValue('git.showGitType', showGitType);
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/import/git/GitSection.tsx:164]: Git type detection and selection needs to happen before you create a git service. This is why the warning pops up without validating anything.

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]

File: frontend/packages/dev-console/src/components/import/devfile/DevfileStrategySection.tsx (hunk 11 above)
Comment (divyanshiGupta) [reply frontend/packages/dev-console/src/components/import/devfile/DevfileStrategySection.tsx:47]: We dont need try-catch here as git-service already handling this.

**Comment by karthikjeeyar** [comment]
UI Breaks when I try using a private git repository (https://github.com/karthikjeeyar/reactApp).
image

**Comment by karthikjeeyar** [comment]
There are two Builder Image detected Alert box shown, is it intentional ? cc: @parvathyvr 
image

**Comment by karthikjeeyar** [comment]
Builder Image Version* dropdown is missing some top margin

image

**Comment by karthikjeeyar** [comment]
I observed one issue while submitting the form, on clicking on submit, then form is again validating the git url and all other fields gets re-rendered. This causes the form footer to go up and down in the page.

**Comment by sanketpathak** [comment]
While QE testing the pr using cluster-bot I found one issue. I entered the git repo as `https://github.com/redhat-developer/devfile-sample`, so it showed me the devfile as the import strategy with other fields auto-populated, then I edited the import strategy as Builder Image with Nodejs as the selection and created the deployment but when I checked annotations on topology I'm seeing that deployment with devfile as import strategy is created instead of the builder image
Screenshot from 2021-08-23 19-42-52

File: frontend/packages/dev-console/src/components/import/ImportStrategySelector.tsx [hunk 24]
```diff
@@ -0,0 +1,84 @@
...
+    },
+  ];
+
+  const onSelect = React.useCallback(
+    (item) => {
+      setFieldValue('import.selectedStrategy.name', item.name);
+      setFieldValue('import.selectedStrategy.type', item.type);
+      setFieldValue('import.selectedStrategy.priority', item.priority);
+      setFieldValue('import.selectedStrategy.detectedFiles', item.detectedFiles);
+      setFieldTouched('import.selectedStrategy', true);
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/import/ImportStrategySelector.tsx:54]: You would also need to update the `build.strategy` here to use proper submit utils for each import strategy.

File: frontend/packages/dev-console/src/components/import/devfile/DevfileStrategySection.tsx [hunk 25]
```diff
@@ -0,0 +1,94 @@
...
+      return t('devconsole~Validated');
+    }
+    if (validated === ValidatedOptions.error) {
+      return t('devconsole~Devfile not detected');
+    }
+    return t(
+      'devconsole~Allows the builds to use a different path to locate your Devfile, relative to the Context Dir field',
+    );
+  }, [t, validated]);
+
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/import/devfile/DevfileStrategySection.tsx:67]: Would be good to run the validation of devfile path once the import strategy is selected.

File: frontend/packages/dev-console/src/components/import/git/DockerSection.tsx [hunk 26]
```diff
@@ -1,29 +1,77 @@
...
+      return t('devconsole~Validated');
+    }
+    if (validated === ValidatedOptions.error) {
+      return t('devconsole~Dockerfile not detected');
+    }
+    return t(
+      'devconsole~Allows the builds to use a different path to locate your Dockerfile, relative to the Context Dir field.',
+    );
+  }, [t, validated]);
+
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/import/git/DockerSection.tsx:48]: Same as devfile section. Should validate once import strategy is selected.

File: frontend/packages/dev-console/src/components/import/git/GitSection.tsx [hunk 27]
```diff
@@ -77,33 +80,104 @@ const GitSection: React.FC<GitSectionProps> = ({
...
+          setFieldValue('devfile.devfileHasError', false);
+        }
+      }
+    },
+    [setFieldValue, values.devfile],
+  );
+
   const handleGitUrlChange = React.useCallback(
     async (url: string, ref: string, dir: string) => {
+      setValidated(ValidatedOptions.default);
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/import/git/GitSection.tsx:124]: Return if form isSubmitting. This is causing re validation when submit is clicked.

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]

**Comment by sahil143** [comment]
@divyanshiGupta Found one issue, after the git URL is validated it detects builder image but doesn't show it. Click on the blank area once, it fluctuates and shows the correct value.

Kapture 2021-08-23 at 23 12 05

File: frontend/packages/dev-console/src/components/import/jar/UploadJar.tsx [hunk 28]
```diff
@@ -65,6 +65,9 @@ const UploadJar: React.FunctionComponent<UploadJarProps> = ({
       tag: tag.name,
       tagObj: tag,
     },
+    import: {
+      showEditImportStrategy: true,
+    },
```
Comment (invincibleJai) [inline frontend/packages/dev-console/src/components/import/jar/UploadJar.tsx:70]: why do we need this here?

**Comment by rottencandy** [comment]
Name field doesn't seem to be getting set for bitbucket urls.

https://user-images.githubusercontent.com/20013884/130504022-8cc38bfb-46b3-41ca-bff5-763816a7044d.mp4

**Comment by divyanshiGupta** [comment]
> Name field doesn't seem to be getting set for bitbucket urls.
> 
>  1.mp4 
> **EDIT:**
> Not specific to bitbucket but appears to happen due to trailing slash in the url.

@saud this issue is in master so we can create a bug for this and then fix it.

**Comment by rottencandy** [comment]
> @saud this issue is in master so we can create a bug for this and then fix it.

Created bug: https://issues.redhat.com/browse/ODC-6276

**Comment by invincibleJai** [comment]
/approve
/label docs-approved
/label px-approved

verified the changes , works as expected

**Comment by divyanshiGupta** [comment]
> @divyanshiGupta Found one issue, after the git URL is validated it detects builder image but doesn't show it. Click on the blank area once, it fluctuates and shows the correct value.
> 
> Kapture 2021-08-23 at 23 12 05



> @divyanshiGupta Found one issue, after the git URL is validated it detects builder image but doesn't show it. Click on the blank area once, it fluctuates and shows the correct value.
> 
> Kapture 2021-08-23 at 23 12 05

I couldn't reproduce it. If you see this again we can create a bug for it as it might be an intermittent issue.

**Comment by rottencandy** [comment]
> I couldn't reproduce it. If you see this again we can create a bug for it as it might be an intermittent issue.
I can see it too, if I enter the url immediately after page loads. Could be happening because builderImages aren't fetched yet.

https://user-images.githubusercontent.com/20013884/130514876-9aa66b36-294d-400c-b23d-b1ef810f8308.mp4

**Comment by divyanshiGupta** [comment]
> > I couldn't reproduce it. If you see this again we can create a bug for it as it might be an intermittent issue.
> 
> I can see it too, if I enter the url immediately after page loads. Could be happening because builderImages aren't fetched yet.
> 
>  0.mp4

Fixed it!

**Review by rohitkrai03 — APPROVED** [review]
/lgtm 

Verified locally. Works as expected.

**Comment by rohitkrai03** [comment]
/label qe-approved

Verified all the below import flows on a cluster created using cluster bot - 
- Import git repo with builder image.
- Import git repo with dockerfile.
- Import git repo with devfile.
- Import git repo with dockerfile but using builder image as import strategy.
- Import builder image sample from sample catalog.
- Import devfile sample from sample catalog.
- Import builder image from developer catalog.
- Import devfile from developer catalog.

Everything works as expected.

**Comment by rohitkrai03** [comment]
/hold

Seems like this PR would need changes from https://github.com/openshift/console/pull/9835 as it changes e2e flows. Sanket is working on rebase his changes on top of this PR. We can get these changes in together.

**Comment by rohitkrai03** [comment]
/lgtm
/hold cancel

Changes from https://github.com/openshift/console/pull/9835 and fixes for breaking e2e tests are now merged into this PR.

**Comment by parvathyvr** [comment]
> There are two Builder Image detected Alert box shown, is it intentional ? cc: @parvathyvr
> image

I think this is a mistake.We do not want to display 2 inline information to the user.We should show only the 1st inline information above the 'Import strategy' section as shown [here](https://docs.google.com/document/d/1yw24SuuraGO6zJQZE8ZCe9KBmIzdPWi3Vyg517aG4TU/edit#heading=h.50rwhkhp8qaw) -The title should be‘Builder Image(s) detected’ and not show the 2nd one we used to have under the 'Builder Image' section. 
As per our new design we have a hover behavior for the star icon as documented [here ](https://docs.google.com/document/d/1yw24SuuraGO6zJQZE8ZCe9KBmIzdPWi3Vyg517aG4TU/edit#heading=h.1j0hl53xhqrk)
@divyanshiGupta

**Comment by rohitkrai03** [comment]
/hold

Merge queue is broken.
